backend/app/usecase/shortlink/updater.go

```diff
@@ -46,21 +53,29 @@ func (u UpdaterPersist) UpdateShortLink(
 		return entity.ShortLink{}, ErrShortLinkNotFound("short link not found")
 	}
 
-	aliasExist, err := u.shortLinkRepo.IsAliasExist(updateArgs.CustomAlias)
-	if err != nil {
-		return entity.ShortLink{}, err
+	newAlias := updateArgs.GetCustomAlias(oldAlias)
+	if newAlias == "" {
+		return entity.ShortLink{}, ErrNewAliasNotSpecified("new alias not specified")
 	}
-	if aliasExist {
-		return entity.ShortLink{}, ErrAliasExist("short link alias already exist")
+
+	// Only check if it exists if user is changing the alias to something else
+	if (newAlias != oldAlias) {
+		aliasExist, err := u.shortLinkRepo.IsAliasExist(newAlias)
+		if err != nil {
+			return entity.ShortLink{}, err
+		}
+		if aliasExist {
+			return entity.ShortLink{}, ErrAliasExist("short link alias already exists")
+		}
```

Comment:
This check can be removed now. We are already checking the existence of short link ownership. Our DB ensures short link exists when mapping for that short link exists.

Reply:
This check is needed because we need to check if the new alias exists. If this check is removed then unknown err will be returned from the API is user attempts to change alias to an alias that already exists.

Reply:
Also, `alias_already_exists` test in `updater_test.go` fails if this check is removed.

Reply:
@Coteh You are right. I misread this!